src/golang/lib/engine/databricks_executor.go

```diff
@@ -0,0 +1,144 @@
+)
+
+// We seperate out the execution step for Databricks Jobs since
+// Databricks takes care of launching Tasks.
+// Steps:
+// 1. Convert each operator into a Task (includes parent dependency).
+// 2. Create a multi-task Job with all the previously created Tasks.
+// 3. Launch job asynchronously
+// 4. Poll on each Task and update accordingly.
+func ExecuteDatabricks(
+	ctx context.Context,
+	workflowDag dag_utils.WorkflowDag,
+	workflowName string,
+	workflowRunMetadata *WorkflowRunMetadata,
+	timeConfig *AqueductTimeConfig,
+	opExecMode operator.ExecutionMode,
+	databricksJobManager *job.DatabricksJobManager,
+) error {
+	inProgressOps := workflowRunMetadata.InProgressOps
+	completedOps := workflowRunMetadata.CompletedOps
+	dag := workflowDag
+
+	// Convert the operators into tasks
+	taskList, err := CreateTaskList(ctx, dag, workflowName, databricksJobManager)
+	if err != nil {
+		return errors.Wrap(err, "Unable to convert operators to Databricks tasks.")
+	}
+
+	// Launch the workflow job with all tasks
+	_, err = databricksJobManager.LaunchMutlipleTaskJob(ctx, workflowName, taskList)
+	if err != nil {
+		return errors.Wrap(err, "Unable to launch workflow job on Databricks.")
+	}
+
+	for _, op := range dag.Operators() {
+		inProgressOps[op.ID()] = op
+	}
+
+	if len(inProgressOps) == 0 {
+		return errors.Newf("No initial operators to schedule.")
+	}
+
+	// Wait a little bit for all active operators to finish before exiting on failure.
+	defer waitForInProgressOperators(ctx, inProgressOps, timeConfig.OperatorPollInterval, timeConfig.CleanupTimeout)
+
+	start := time.Now()
+
+	for len(inProgressOps) > 0 {
+		if time.Since(start) > timeConfig.ExecTimeout {
+			return errors.New("Reached timeout waiting for workflow to complete.")
+		}
+
+		for _, op := range inProgressOps {
+			// Poll on the individual operator
+			execState, err := op.Poll(ctx)
+			logrus.Infof("execState of operator:%v", op.JobSpec().JobName())
+			logrus.Info(execState.Status)
+			if err != nil {
+				return err
+			}
+
+			if execState.Status == shared.PendingExecutionStatus {
+				continue
+			} else if execState.Status == shared.RunningExecutionStatus {
+				continue
+			}
+
+			if !execState.Terminated() {
+				return errors.Newf("Internal error: the operator is expected to have terminated, but instead has status %s", execState.Status)
+			}
+
+			// From here on we can assume that the operator has terminated.
+			if opExecMode == operator.Publish {
+				err = op.PersistResult(ctx)
+				if err != nil {
+					return errors.Wrapf(err, "Error when finishing execution of operator %s", op.Name())
+				}
+			}
+
+			// Add the operator to the completed stack, and remove it from the in-progress one.
+			if _, ok := completedOps[op.ID()]; ok {
+				return errors.Newf("Internal error: operator %s was completed twice.", op.Name())
+			}
+			completedOps[op.ID()] = op
+			delete(inProgressOps, op.ID())
+
+			time.Sleep(timeConfig.OperatorPollInterval)
```

Comment:
Is this intentional to wait this interval for every single operator? I'm imagining we can do a batch poll (and even export if databrick has API to do this), wait for some interval, and do another batch poll

Reply:
I think it's more critical if we need a good balance of execution time and rate limit. If neither is a problem we can solve it later

Reply:
cc @kenxu95 I think so! https://github.com/aqueducthq/aqueduct/blob/main/src/golang/lib/engine/aq_engine.go